File: lab3/c_factory/parrot.c
#include "parrot.h"
#include <stdio.h>

Parrot* create(const char* name){
    Parrot* p = (Parrot*) malloc(sizeof(Parrot));
    construct(p, name);
    return p;
}

void construct(Parrot* p, const char* name){
    p->vtable = parrot_vtable;
    p->name = name;
}

const char* name(Parrot* p){
    return p->name;
}

const char* greet(){
    return "sqwak";
}

const char* menu(){
    return "voće";
}

int size(){
    return sizeof(Parrot);
}

File: lab3/factory_java/src/main/java/hr/fer/zemris/ooup/lab2/model/plugins/Parrot.java
package hr.fer.zemris.ooup.lab2.model.plugins;

import hr.fer.zemris.ooup.lab2.model.Animal;

public class Parrot extends Animal {

    private String animalName;

    public Parrot(String name){
        super();
        this.animalName = name;
    }

    @Override
    public String name() {
        return animalName;
    }

    @Override
    public String greet() {
        return "sqwak";
    }

    @Override
    public String menu() {
        return "voće";
    }
}
